### PR TITLE
feat(plugman): support framework link attribute 

### PR DESCRIPTION
**Pull request by TylerBreau**
Update is backwards compatible with previous embed behaviour.

### Platforms affected
iOS

### Motivation and Context

Currently, the framework tag does not support both linking and embedding a framework. Comments in the code suggest this was a limitation in Apple's code.

This limitation no longer exists. The current default behaviour of XCode (version 14) when dragging and dropping an XCFramework into General -> "Frameworks, Libraries, and Embedded Content" is to both embed and link the XCFramework.
I have only tested this with XCFramework but I assume other kinds of frameworks are the same.

This change allows the framework tag to both link and embed a framework.

In my real-world use case, I am using custom XCFrameworks to get c binaries into my cordova app.
If the XCFramework is not embedded then the App will crash during runtime because it can not find the binaries. This is expected because the binaries were not embedded into the app.
If the XCFramework is not linked then the App can not be built due to linker errors. This is expected because the binaries are not being linked to the app.

cordova-ios unit tests use cordova-common directly. As a result, this PR depends on https://github.com/apache/cordova-common/pull/181 (cordova-common 4.1.0).
In actual usage, cordova-cli parses the framework attributes and gives them to cordova-ios. As a result, **this PR also depends on a cordova-cli that is using https://github.com/apache/cordova-common/pull/181 (cordova-common 4.1.0)**

cordova-docs pr: https://github.com/apache/cordova-docs/pull/1267

### Description

Add code that checks for new framework attribute "link".
If link is not explicitly set, fallback to the previous behaviour (link = !embed).

### Testing

I've ran npm test.

Tested various combinations of framework tag's embed and link attributes by installing a plugin using these attributes. I would observe how the XCFramework is added to the the XCode project with the following focuses:
-Is the XCFramework embedded (or not embedded) in Build Phrases?
-Is the XCFramework linked (or not linked) in Build Phrases?
-Is the XCFramework set to "Embed and Sign" in General?

I have ran my company's app using these changes.

### Checklist

- [x] I've run the tests to see all new and existing tests pass
- [x] I added automated test coverage as appropriate for this change
- [x] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [x] I've updated the documentation if necessary

### CODE REVIEW

**Comment by TylerBreau** [comment]
This change spans across 5 repositories. These are the 5 PRs:

- https://github.com/apache/cordova-common/pull/179
- https://github.com/apache/cordova-lib/pull/898
- https://github.com/apache/cordova-cli/pull/584
- https://github.com/apache/cordova-ios/pull/1266
- https://github.com/apache/cordova-docs/pull/1267

**Comment by msmtamburro** [comment]
Nice, thank you.

File: package.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
     "node": ">=10"
   },
   "dependencies": {
-    "cordova-common": "^4.0.2",
+    "cordova-common": "file:cordova-common-4.0.3-dev.tgz",
```
Comment (breautek) [inline package.json:44]: cordova-common has been published now, this should be pinned to `^4.1.0`

**Comment by TylerBreau** [comment]
Opening PR for review.

Noting again, this PR requires a cordova-cli that is using cordova-common@4.1.0. If your environment does not contain this, cordova-ios will not receive the link attribute in an actual cordova project.
Unit tests will work regardless because they use cordova-common directly.

**Review by breautek — APPROVED** [review]

**Review by dpogue — APPROVED** [review]

**Comment by TylerBreau** [comment]
Forced-push to update branch with master.

**Comment by breautek** [comment]
The failing test does not appear to be related to this PR.

We've investigated and determined that issue happens in current master on NodeJS 18. I've created a bug ticket with further details: https://github.com/apache/cordova-ios/issues/1290

**Comment by breautek** [comment]
Waiting on https://github.com/apache/cordova-ios/pull/1302 so that we can confirm if all test passes.

**Pull request merged**